cranelift-codegen/src/isa/x86/enc_tables.rs

```diff
@@ -16,9 +16,20 @@ use crate::isa::{self, TargetIsa};
 use crate::predicates;
 use crate::regalloc::RegDiversions;
 
+use cranelift_codegen_shared::isa::x86::EncodingBits;
+
 include!(concat!(env!("OUT_DIR"), "/encoding-x86.rs"));
 include!(concat!(env!("OUT_DIR"), "/legalize-x86.rs"));
 
+/// Whether the REX prefix is needed for encoding extended registers (via REX.R/B).
+///
+/// Normal x86 instructions have only 3 bits for encoding a register.
+/// The REX prefix adds REX.R and REX.B bits, interpreted as fourth bits.
+pub fn needs_rex_rb(reg: RegUnit) -> bool {
```

Comment:
small quibble: this PR doesn't touch REX prefixes for instructions with memory operands, but this is also a correct check for if the index register in a memory operand needs `X`. How do you feel about naming this simply `is_extended_reg`?